File: feedback_controller/fbc/neural_network/kinematics/generated_pycode/Jpos_head_HEAD.py
# HEAD POINT POSITION JACOBIAN
# Try: [J,~]=Jpos_head_HEAD.compute_jac(0,0, 0,0,0,0,0,0,0, 0)
import numpy as np

class Jpos_head_HEAD:
	@staticmethod
	def compute_jac(t1, t2, h1, h2):
		J = np.zeros([3,4]);
		J[0,0] = 0.035*np.cos(t1)*np.sin(h1) - 0.3945*np.sin(t1)*np.sin(t2) + 0.035*np.cos(h1)*np.cos(t2)*np.sin(t1)
		J[0,1] = 0.3945*np.cos(t1)*np.cos(t2) + 0.035*np.cos(h1)*np.cos(t1)*np.sin(t2)
		J[0,2] = 0.035*np.cos(h1)*np.sin(t1) + 0.035*np.cos(t1)*np.cos(t2)*np.sin(h1)
		J[0,3] = 0.0
		J[1,0] = 0.3945*np.cos(t1)*np.sin(t2) + 0.035*np.sin(h1)*np.sin(t1) - 0.035*np.cos(h1)*np.cos(t1)*np.cos(t2)
		J[1,1] = 0.3945*np.cos(t2)*np.sin(t1) + 0.035*np.cos(h1)*np.sin(t1)*np.sin(t2)
		J[1,2] = 0.035*np.cos(t2)*np.sin(h1)*np.sin(t1) - 0.035*np.cos(h1)*np.cos(t1)
		J[1,3] = 0.0
		J[2,0] = 0.0
		J[2,1] = 0.035*np.cos(h1)*np.cos(t2) - 0.3945*np.sin(t2)
		J[2,2] = -0.035*np.sin(h1)*np.sin(t2)
		J[2,3] = 0.0
		return J
	#end of compute Jacobian
#end of class Jpos_head_HEAD
def main():
	print(Jpos_head_HEAD.compute_jac(0,0, 0,0,0,0, 0,0,0, 0))

if __name__ == '__main__':
	main()

File: feedback_controller/fbc/neural_network/kinematics/generated_code/Jpos_head_HEAD.m
% HEAD POINT POSITION JACOBIAN
function J = Jpos_head_HEAD(t1, t2, h1, h2)
    J(1,1) = 0.035*cos(t1)*sin(h1) - 0.3945*sin(t1)*sin(t2) + 0.035*cos(h1)*cos(t2)*sin(t1);
    J(1,2) = 0.3945*cos(t1)*cos(t2) + 0.035*cos(h1)*cos(t1)*sin(t2);
    J(1,3) = 0.035*cos(h1)*sin(t1) + 0.035*cos(t1)*cos(t2)*sin(h1);
    J(1,4) = 0.0;
    J(2,1) = 0.3945*cos(t1)*sin(t2) + 0.035*sin(h1)*sin(t1) - 0.035*cos(h1)*cos(t1)*cos(t2);
    J(2,2) = 0.3945*cos(t2)*sin(t1) + 0.035*cos(h1)*sin(t1)*sin(t2);
    J(2,3) = 0.035*cos(t2)*sin(h1)*sin(t1) - 0.035*cos(h1)*cos(t1);
    J(2,4) = 0.0;
    J(3,1) = 0.0;
    J(3,2) = 0.035*cos(h1)*cos(t2) - 0.3945*sin(t2);
    J(3,3) = -0.035*sin(h1)*sin(t2);
    J(3,4) = 0.0;
%end of Jacobian computation

